common/flatpak-oci-registry.c

```diff
@@ -3227,13 +3248,22 @@ add_image_to_appstream (FlatpakHttpSession        *http_session,
       component = next;
     }
 
+  certificates = flatpak_get_certificates_for_uri (index_uri, &error);
```

Comment:
Can you push this certification load to flatpak_oci_index_make_appstream(). Otherwise we're going to repeat this for the same uri for every app in the index.